src/corelib/Networking/v2/Layer3/SecurityGroupRule.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+namespace OpenStack.Networking.v2.Layer3
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class SecurityGroupRule
+    {
+        /// <summary>
+        ///ngress or egress: the direction in which the security group rule is applied.
+        ///For a compute instance, an ingress security group rule is applied to incoming (ingress) traffic for that instance. 
+        ///An egress rule is applied to traffic leaving the instance.
+        /// </summary>
+        [JsonProperty("direction")]
+        public string Direction;
```

Comment:
This should be a subclass of `OpenStack.Serialization.StringEnumeration`, see `OpenStack.Compute.v2_1.AddressType` for an example.

When a resource property has a fixed set of values, it helps to wrap that in a enum-like class so that user's don't have to go to the API documentation to know what the allowed values are, in this case "egress" and "ingress". The hierarchy of the `StringEnumeration` type enables the SDK to define well known values while still allowing for provider specific values to be defined later. For example Rackspace may have different values available in their SDK which builds on top of OpenStack.NET.